providers/GraphQLProvider.tsx

```diff
@@ -0,0 +1,54 @@
+import { createClient, Provider, dedupExchange, cacheExchange, fetchExchange } from "urql";
+import { authExchange } from '@urql/exchange-auth';
+import { PropsWithChildren } from "react";
+
+import useApp from "../hooks/useApp";
+
+interface IAuthState {
+  token: string;
+}
+
+const GraphQLProvider: React.FC<PropsWithChildren<{}>> = (props) => {
+  const { domain, token } = useApp()?.getState() || {};
+  const client = createClient({
+    url :`https://${domain}/graphql/`,
+    exchanges: [
+      dedupExchange,
+      cacheExchange,
+      authExchange<IAuthState>({
+        addAuthToOperation: ({ authState, operation }) => {
+          if (!authState || !authState?.token) {
+            return operation;
+          }
+
+          const fetchOptions =
+            typeof operation.context.fetchOptions === 'function'
+              ? operation.context.fetchOptions()
+              : operation.context.fetchOptions || {};
+
+          return {
+            ...operation,
+            context: {
+              ...operation.context,
+              fetchOptions: {
+                ...fetchOptions,
+                headers: {
+                  ...fetchOptions.headers,
+                  "Authorization-Bearer": authState.token,
+                },
+              },
+            },
+          };
+        },
+        getAuth: async ({ authState, mutate }) =>  {
```

Comment:
@zaiste jwm0 this is a place, that we need to figure out how app-bridge based JWT tokens are rotated.

Reply:
If we just call `getState()` here we'll always get fresh token

Reply:
I've just checked (using `setInterval`) and on the same app (meaning I cannot run `useApp` outside of function component), I tried to
```
console.log(app.getState().token)
```
and it was always the same and eventually expired

Reply:
I've briefly checked dashboard and app-bridge and haven't found any means to refresh this token, when needed.